service/history/replicatorQueueProcessor.go

```diff
@@ -219,29 +219,55 @@ func (p *replicatorQueueProcessorImpl) processHistoryReplicationTask(
 	}
 
 	err = p.replicator.Publish(replicationTask)
-	if err == messaging.ErrMessageSizeLimit && replicationTask.GetHistoryTaskAttributes() != nil {
+	if err == messaging.ErrMessageSizeLimit {
 		// message size exceeds the server messaging size limit
 		// for this specific case, just send out a metadata message and
 		// let receiver fetch from source (for the concrete history events)
-		err = p.replicator.Publish(p.generateHistoryMetadataTask(replicationTask.GetHistoryTaskAttributes().TargetClusters, task))
+		if metadataTask := p.generateHistoryMetadataTask(
+			task,
+			replicationTask,
+		); metadataTask != nil {
+			err = p.replicator.Publish(metadataTask)
+		}
 	}
 	return err
 }
 
-func (p *replicatorQueueProcessorImpl) generateHistoryMetadataTask(targetClusters []string, task *persistenceblobs.ReplicationTaskInfo) *replicationgenpb.ReplicationTask {
-	return &replicationgenpb.ReplicationTask{
-		TaskType: enumsgenpb.REPLICATION_TASK_TYPE_HISTORY_METADATA_TASK,
-		Attributes: &replicationgenpb.ReplicationTask_HistoryMetadataTaskAttributes{
-			HistoryMetadataTaskAttributes: &replicationgenpb.HistoryMetadataTaskAttributes{
-				TargetClusters: targetClusters,
-				NamespaceId:    task.GetNamespaceId(),
-				WorkflowId:     task.GetWorkflowId(),
-				RunId:          task.GetRunId(),
-				FirstEventId:   task.GetFirstEventId(),
-				NextEventId:    task.GetNextEventId(),
+func (p *replicatorQueueProcessorImpl) generateHistoryMetadataTask(
+	task *persistenceblobs.ReplicationTaskInfo,
+	replicationTask *replicationgenpb.ReplicationTask,
+) *replicationgenpb.ReplicationTask {
+
+	if replicationTask.GetHistoryTaskAttributes() != nil {
+		return &replicationgenpb.ReplicationTask{
+			TaskType: enumsgenpb.REPLICATION_TASK_TYPE_HISTORY_METADATA_TASK,
+			Attributes: &replicationgenpb.ReplicationTask_HistoryMetadataTaskAttributes{
+				HistoryMetadataTaskAttributes: &replicationgenpb.HistoryMetadataTaskAttributes{
+					TargetClusters: replicationTask.GetHistoryTaskAttributes().GetTargetClusters(),
+					NamespaceId:    task.GetNamespaceId(),
+					WorkflowId:     task.GetWorkflowId(),
+					RunId:          task.GetRunId(),
+					FirstEventId:   task.GetFirstEventId(),
+					NextEventId:    task.GetNextEventId(),
+					Version:        common.EmptyVersion,
```

Comment:
This is the only one change I had to made manually. In cadence it used to be `nil` but we already use `EmptyVerison` const in many places.